src/main.py

```diff
@@ -0,0 +1,78 @@
+import asyncio
+import argparse
+
+from pipeline.eval import eval_on_records
+from utils import validate_steps
```

Comment:
I feel like `main.py` shouldn't be a part of `src` actually. It should likely be moved out of `src`. So should be `requirements.txt`.